e2e/scenario/scenario.go

```diff
@@ -0,0 +1,154 @@
+		integrationImageTag:        integrationImageTag,
+		podOptionalNetworkSchema:   podOptionalNetworkSchema(k8sServerInfo),
+		nodeOptionalNetworkSchema:  nodeOptionalNetworkSchema(k8sServerInfo, unprivileged),
+	}
+}
+
+func (s Scenario) String() string {
+	str := fmt.Sprintf(
+		"rbac=%v,ksm-instance-one.rbac.create=%v,ksm-instance-one.image.tag=%s,daemonset.unprivileged=%v,daemonset.image.repository=%s,daemonset.image.tag=%s",
+		s.rbac,
+		s.rbac,
+		s.ksmVersion,
+		s.unprivileged,
+		s.integrationImageRepository,
+		s.integrationImageTag,
+	)
+	if s.twoKSMInstances {
+		return fmt.Sprintf(
+			"%s,ksm-instance-two.rbac.create=%v,ksm-instance-two.image.tag=%s,two-ksm-instances=true",
+			str,
+			s.rbac,
+			s.ksmVersion,
+		)
+	}
+	return str
+}
+
+// GetSchemasForJob returns the json schemas that should be use to
+// match the test scenario.
+func (s Scenario) GetSchemasForJob(job string) jsonschema.EventTypeToSchemaFilename {
+
+	eventTypeSchemas := defaultEventTypeToSchemaFilename()
+
+	if s.nodeOptionalNetworkSchema {
+		eventTypeSchemas["kubelet"]["K8sNodeSample"] = "node-no-network.json"
+	}
+
+	if s.podOptionalNetworkSchema {
+		eventTypeSchemas["kubelet"]["K8sPodSample"] = "pod-no-network.json"
+	}
+
+	return eventTypeSchemas[string(job)]
+}
+
+func defaultEventTypeToSchemaFilename() map[string]jsonschema.EventTypeToSchemaFilename {
+	return map[string]jsonschema.EventTypeToSchemaFilename{
+		"kube-state-metrics": {
+			"K8sReplicasetSample":  "replicaset.json",
+			"K8sNamespaceSample":   "namespace.json",
+			"K8sDeploymentSample":  "deployment.json",
+			"K8sDaemonsetSample":   "daemonset.json",
+			"K8sStatefulsetSample": "statefulset.json",
+			"K8sEndpointSample":    "endpoint.json",
+			"K8sServiceSample":     "service.json",
+		},
+		"kubelet": {
+			"K8sPodSample":       "pod.json",
+			"K8sContainerSample": "container.json",
+			"K8sNodeSample":      "node.json",
+			"K8sVolumeSample":    "volume.json",
+			"K8sClusterSample":   "cluster.json",
+		},
+		"scheduler": {
+			"K8sSchedulerSample": "scheduler.json",
+		},
+		"etcd": {
+			"K8sEtcdSample": "etcd.json",
+		},
+		"controller-manager": {
+			"K8sControllerManagerSample": "controller-manager.json",
+		},
+		"api-server": {
+			"K8sApiServerSample": "apiserver.json",
+		},
+	}
+}
+
+// nodeOptionalNetworkSchema returns true when kubernetes version is 1.18
+// or newer and unprivileged is true.
+func nodeOptionalNetworkSchema(k8sServerInfo *version.Info, unprivileged bool) bool {
+	if k8sServerInfo == nil {
+		return false
+	}
+	major, err := strconv.Atoi(k8sServerInfo.Major)
+	if err != nil {
+		return false
+	}
+	if major > 1 {
+		return unprivileged
+	}
+	minor, err := strconv.Atoi(k8sServerInfo.Minor)
+	if err != nil {
+		return false
+	}
+	return minor >= 18 && unprivileged
+}
+
+// podOptionalNetworkSchema returns true when kubernetes version if 1.18
+// or newer.
+func podOptionalNetworkSchema(k8sServerInfo *version.Info) bool {
```

Comment:
The changelog entry or PR description doesn't mention anything about pod's network metrics.

Reply:
Yeah I was trying something that was getting out of hand and I forgot to delete this